lambda/query-user-services/tests/query-user-services.test.ts

```diff
@@ -0,0 +1,256 @@
+    }).toThrow(ValidationError);
+  });
+  test("throws error when component_id is missing", () => {
+    const txmaEvent = parseRecordBody(
+      JSON.stringify({
+        services: [
+          {
+            client_id: "client_id",
+            timestamp: new Date().toISOString,
+            event_name: "event_name",
+            user: {
+              user_id: "user_id",
+            },
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrow(ValidationError);
+  });
+  test(" throws error when user is missing", () => {
+    const txmaEvent = parseRecordBody(
+      JSON.stringify({
+        services: [
+          {
+            client_id: "client_id",
+            timestamp: new Date().toISOString,
+            event_name: "event_name",
+            component_id: "component_id",
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrow(ValidationError);
+  });
+  test("throws error when user_id  is missing", () => {
+    const txmaEvent = parseRecordBody(
+      JSON.stringify({
+        services: [
+          {
+            client_id: "client_id",
+            timestamp: new Date().toISOString,
+            event_name: "event_name",
+            component_id: "component_id",
+            user: {},
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrow(ValidationError);
+  });
+});
+
+describe("validateUser", () => {
+  test("doesn't throw an error with valid user data", () => {
+    expect(validateUser(user)).toBe(undefined);
+  });
+});
+
+describe("sendSqsMessage", () => {
+  beforeEach(() => {
+    sqsMock.reset();
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  test("Send the SQS event on the queue", async () => {
+    sqsMock.on(SendMessageCommand).resolves({ MessageId: MOCK_MESSAGE_ID });
+    const messageId = await sendSqsMessage(Object, MOCK_QUEUE_URL);
+    expect(messageId).toEqual(MOCK_MESSAGE_ID);
+    expect(
+      sqsMock.commandCalls(SendMessageCommand, {
+        QueueUrl: MOCK_QUEUE_URL,
+        MessageBody: JSON.stringify(Object),
+      })
+    );
+  });
+});
+
+describe("handler", () => {
+  const serviceList: Service[] = [
+    {
+      client_id: "client_id",
+      count_successful_logins: 2,
+      last_accessed: new Date(),
+    },
+  ];
+  beforeEach(() => {
+    dynamoMock.reset();
+
+    process.env.TABLE_NAME = "TABLE_NAME";
+    sqsMock.on(SendMessageCommand).resolves({ MessageId: MOCK_MESSAGE_ID });
+    dynamoMock.on(GetCommand).resolves({ Item: serviceList });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  test("Queries the dynamo db and send an sqs event", async () => {
+    await handler(TEST_SQS_EVENT);
+    expect(sqsMock.commandCalls(SendMessageCommand).length).toEqual(2);
+    expect(dynamoMock.commandCalls(GetCommand).length).toEqual(2);
```

Comment:
Could also check what we're sending there with `toHaveReceivedCommandWith`?
Or you thinking that's covered off in the rest of your test suite?

Reply:
Good shout I will include that